\documentclass{report}

% set font encoding for PDFLaTeX, XeLaTeX, or LuaTeX
\usepackage{ifxetex,ifluatex}

\if\ifxetex T\else\ifluatex T\else F\fi\fi T%
  \usepackage{fontspec}
\else
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{lmodern}
\fi

\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{bm}
\usepackage{bbm}
\usepackage{mathtools}
\usepackage{physics}

\usepackage{enumitem}
\usepackage{multicol}
\usepackage{graphicx}
\usepackage[dvipsnames]{xcolor}

\usepackage{hyperref}
\hypersetup{colorlinks=true,}

\usepackage[parfill]{parskip}
\usepackage{lipsum}
\usepackage[export]{adjustbox}
\usepackage{listings}

\usepackage{xparse} 
\usepackage{subfig} 
\usepackage{xparse} 
\usepackage{float}

\usepackage[sorting=none]{biblatex} 

%%%%%This is an image table command, can likely be deleted
\newcommand{\subf}[2]{

%
{\small 
\begin{tabular}
  [t]{@{}c@{}} #1\ 
  \#2 
\end{tabular}
}

%
} 

\makeatletter
\renewcommand*\env@matrix[1][c]{\hskip -\arraycolsep
  \let\@ifnextchar\new@ifnextchar
  \array{*\c@MaxMatrixCols #1}}
\makeatother
%%%%%% Tensor Product
\NewDocumentCommand{\tens}{e{_^}}{ 
\mathbin{\mathop{\otimes}\displaylimits \IfValueT{#1}{_{#1}} \IfValueT{#2}{^{#2}} }}
%%%%%% Add \R Reals
\newcommand{\R}{\mathbb{R}} 
\newcommand{\N}{\mathbb{N}} 
\newcommand{\Z}{\mathbb{Z}} 
%%%%%% Missing citation warn
\newcommand{\CITEMISSING}{\colorbox{BurntOrange}{CITE ME}}
%%%%%% Add \theorem float
\newtheorem{theorem}{Theorem}
%%%%%% Add \definition float
\theoremstyle{definition} 
\newtheorem{definition}{Definition}[section]
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%Uncomment to add citation library 
\bibliography{lib} 
\title{Notes}
\author{David Helekal}
\begin{document}
\maketitle
\newpage
\tableofcontents
\newpage
\chapter{Introduction}
\section{Motivation}
In epidemiology, it is often desired to be able to reconstruct the history of a pathogen population and it's structure. The problem of reconstructing the hisotry of a pathogen population can be tackles using phylodynamics. Phylodynamics utilises genomic data to assemble phylogenies, which are then used to infer the population size history. This is possible by viewing a phylogeny as a realisation of a coalescent process, wit appropriately rescaled time. This claim can be justified by viewing the coalescent as a Moran model, run backwards in time with the time rate equal to the population size \cite{griffiths_sampling_1994}.\\
Within this report we will first introduce the coalescent process for phylodynamic inference, review it's inhomogenous generalisation, and finally introduce the main result of this work, a new model capable of doing local phylodynamic inference, i.e. on a subset of the whole population capable of detecting and modelling clonal expansions.\\
Clonal expansions are a process in which a particular subsest of a given bacterial strain undergoes explosive population growth that can be traced back to a particular individual \cite{smith_how_1993}. The presence of clonal expansions in bacterial populations have been of long-standing interest and is implicated in epidemic processes, were an outbreak can be traced to a single ancestor \cite{smith_how_1993,spratt_displaying_2004,fraser_neutral_2005,ledda_re-emergence_2017}. This often happens when a particular strain or individual obtains a variant of a particular gene that confers evolutionary advantange, for example, antibiotic resistance \cite{holden_genomic_2013,hsu_evolutionary_2015, ledda_re-emergence_2017}.\\
The presence of clonal expansions leaves an imprint in the overall population structure of a given bacterial strain, the particular topology associated with this often being referred to as star-like \cite{smith_how_1993,spratt_displaying_2004}.
The problem of detecting hidden population structure corresponding to clonal expansions has become a problem of interest in epidemiology and outbreak surveillance \cite{volz_identification_nodate}.\\
While methods to detect inhomogeneities in the population structure and size have been of interest since the early days of genetic sequencing \cite{smith_how_1993,spratt_displaying_2004}, the interest in the problem increased with whole genome sequencing becoming more accessible and affordable \cite{holden_genomic_2013,dearlove_measuring_2015,eldholm_four_2015}.\\
Despite the problems of inferring population size from a genealogy and detecting heterogeneities in the population size of the entire population being intrinsically tied, all but one method \cite{volz_identification_nodate}, to our knowledge, rely either on manual detection or indirect detection. We aim to propose a simulation for the formation of clonal expansions in genealogy using the structured coalescent process, and devise a fully bayesian method for joint estimation and detection of relative population size and clonal expansions.
\section{Existing Work}
\subsection{Phylodynamic Methods}
A phylogeny is a labeled tree, where leaf nodes correspond to sampling events, and internal nodes to the divergence of two separate lineages from a common ancestor. Branch length labels correspond to time. As such a phylogeny can be viewed as a realisation of Kingman's Coalescent process which shall now be introduced, and the justification behind why viewing a phylogeny as its realisation briefly outlined.\\
Kingman's Coalescent is a continuous time markov chain stochastic process, defined on the statespace ${1,2,3, ... ,K}$ which can be interpreted as a set of $j$ particles, where each pair of particles independently coalesces at a constant rate. This gives transition rates:
\begin{gather}
g(j, j-1) = \binom{j}{2}\lambda\quad\lambda\in\R^+
\end{gather}
\cite{kingman_coalescent_1982}\\
By taking a backwards in time approximation of the Wright-Fisher model, the coalescent process can be modified to model evolution of genealogies, i.e. how do the ancestors of a set of individuals relate to each other backwards in time. Denote the relative population size at time $t$ by $\alpha(t)$. Under such modification the transition rates become:
\begin{gather}
g(j, j-1) = \binom{j}{2}\cdot\frac{1}{\alpha(t)}
\end{gather}
\cite{griffiths_sampling_1994}\\
One way to interpret this is as a rescaling of time inversely proportional to the population size under Wright-Fisher model \cite{hein_gene_2004}.\\
As the transition rates depend on the relative population size, it is possible to utilise this model for the inverse problem of determining the history of the size of a population, based on genealogies reconstructed from genomic samples.
Such methods are often referred to as SkyGrid methods, have been first introduced in \cite{pybus_integrated_2000} and \cite{drummond_estimating_2002}.\\
The framework has then been extended to allow for piecewise continuous population size functions, referred to as SkyGrid \cite{gill_improving_2013} and to include covariates \cite{gill_understanding_2016}.
\subsection{Local Phylodynamics}
Whereas phylodynamics considers the size history of an entire population, the idea behind local phylodynamics is to consider the histories of selected subsets of related individuals, where each subset can be traced to a single ancestor. This can for example be practical when trying to detect hidden outbreaks, or clonal expansions due to a particular strain of a bacteria gaining a fitness advantage. This idea has been investigated in \cite{volz_identification_nodate}, where a null hypothesis based testing framework is developed to identify subsets of individuals that seem to have a significantly different opulation history than the the rest of the phylogeny. Recently, a birth-death type model that allows for heterogenous growth rate parameters has been introduced in \cite{barido-sottani_multitype_2020}.
\chapter{Methods}
\section{Coalescent Preliminaries}
The coalescemt process can be characterised as a time-inhomogenous pure-death markov process. We now conduct a brief analysis of the process and re-derive some properties.\\
The waiting times can be derived as follows.
For an inhomogenous CTMC, let $E_j(t)$ be the total exit rate from state $j$ at time $t$.
By the markov property individual exit events from a given state only depend on the state and given time, i.e. they form a time-inhomogenous poisson process.
As such the probability of no events in an interval $[t,t+s]\quad s\in \R^+$ is 
\begin{gather}
\exp(-\int_t^{t+s}E_j(\tau)d\tau) = \exp(-\int_0^{s}E_j(t+\tau)d\tau)
\end{gather}
The waiting times are defined as
\begin{gather}
W_j(t) = \inf\{s:X(t+s)\neq j \mid X(t) = j\}
\end{gather}
As such
\begin{gather}
W_j(t) > s \Rightarrow \forall \tau\in[t, t+s]\quad X(\tau) = j
\end{gather}
Furthermore the above relation holds iff no exit event have occured in the time interval $[t,t+s]$. As such:
\begin{align*}
&P[W_j(t) > s] = P[\text{no exit events in }[t,t+s]] = \exp(-\int_0^{s}E_j(t+\tau)d\tau)\\
&P[W_j(t) < s] = 1 - \exp(-\int_0^{s}E_j(t+\tau)d\tau)
\end{align*}
In the case of phylodynamic coalescent this becomes
\begin{gather}\label{eq:w_t}
P[W_j(t) \leq s] = 1 - \exp(-\int_0^{s}\frac{\binom{j}{2}}{\alpha(t+\tau)}d\tau)
\end{gather}
From equation \ref{eq:w_t} we can see that the waiting times between individual coalescent events depend on the relative population size.
\section{Simulating Phylogenies}
We are interested in simulating the scenario where sampling events and the relative population size is given. Denote the collection of sampling events in decreasing time order by:
\begin{gather}
\{s_j\}_{1\leq j\leq K}, \quad i>l \Leftrightarrow s_i \leq s_l\quad \forall i,l
\end{gather}
Under this scenario, phylogenies can be simulated as realisations of the inhomogenous coalescent process, conditioned on sampling events, using modified Gillespie's Algorithm \cite{erban_practical_nodate}. The key difference between standard Gillespie's Algorithm setting and our problem setting is that we need to condition on the sampling events.\\
Hence at any time greater than the earliest sampling event $t_i > s_K$ in the simulation, we first need to determine whether an event happens or not in the interval $[s_j, t_i]$ with $s_j = max\left\{s\in S : s<t_i\right\}$. Define $\Delta t_i \triangleq t_i-s_j$, and assume there are $j$ individual lineages extant at time $t$. We are interested in the probability
\begin{gather*}
  P[W_j(t_i) > \Delta t_i] = \exp(-\int_0^{\Delta t_i}\frac{\binom{j}{2}}{\alpha(t+\tau)}d\tau)
\end{gather*}
To determine the next in simulation draw $r \sim \texttt{U}([0,1])$. 
If $r < P[W_j(t_i) > \Delta t_i]$, no events happen in the interval $[s_j, t_i]$. As such set $t_{i+1} = s_j$ and repeat the process.\\
If $r > P[W_j(t_i) > \Delta t_i]$ a coalescent event must happen in the interval $[s_j, t_i]$. Therefore we need to determine the next waiting time $W_j(t_i)$. The cumulative distribution function (CDF) of $W_j(t_i)$ is given by
\begin{gather}
  P[W_j(t_i) < c \mid W_j(t_i) < \Delta t_i] =\frac{1-\exp(-\int_{0}^{c}{\frac{\binom{j}{2}}{\alpha(t+\tau)}d\tau})}{1-P[W_j(t_i) > \Delta t_i]}
\end{gather}
$W_j(t_i)$ can be generated by the inverse transform of exponentially distributed variables. Let $W \sim \texttt{exp}(1)$ and note that random variates $W'$ with the CDF
\begin{gather}
  P[W' \leq c] = P[W\leq c\mid W\leq\Delta' t] \quad\forall c \leq \Delta t
\end{gather}
can be generated from random variables $u \sim \texttt{U}([0,1])$ by the following inverse transform
\begin{gather}
T(u) = -1\log[1-u(1-\exp(-\Delta' t))]
\end{gather} 
This can be justified as follows
\begin{gather*}
  P[T(u)\leq c] = P[u\leq T^{-1}(c)]
\end{gather*}
Using the property that $P[u\leq a] = a$ and requiring that $T(u)$ follows the same distribution as $W'$ we obtain
\begin{align*}
  &\Rightarrow& P[u\leq T^{-1}(c)] &= \frac{\int_0^c\exp(- t)dt}{\int_0^{\Delta t}\exp(- t)dt}\\
  &\Rightarrow& T^{-1}(c) &= \frac{1-\exp(- c)}{1-\exp(- \Delta t)}
\end{align*}
Being abled to generate appropriate random variates $W'$, a further inverse transform is applied, this time to obtain $W_j(t)$ from $W'$. First, we require that $\Delta' t$ be such that $P[W < \Delta' t]$ is equal to $P[W_j(t_i) < \Delta t]$. Next, we seek a function $G(W';t)$ such that
\begin{gather}
\begin{aligned}
&&P[W' \leq G^{-1}(c;t)] &= P[W_j(t) \leq c \mid W_j(t)\leq\Delta t]\\
&\Rightarrow& \frac{1-\exp(-G^{-1}(c;t))}{1-P[W > \Delta't]} &= \frac{1-\exp(-\int_{0}^{c}{\frac{\binom{j}{2}}{\alpha(t+\tau)}d\tau})}{1-P[W_j(t_i) > \Delta t_i]}
\end{aligned}
\end{gather}
Using that $1-P[W > \Delta't] = 1-P[W_j(t_i) > \Delta t_i]$ we obtain
\begin{align}
&\Rightarrow& 1-\exp(-G^{-1}(c;t)) &= 1-\exp(-\int_{0}^{c}{\frac{\binom{j}{2}}{\alpha(t+\tau)}d\tau})\\
&\Rightarrow& G^{-1}(c;t) &= \int_{0}^{c}{\frac{\binom{j}{2}}{\alpha(t+\tau)}d\tau}\label{eq:inv_int}
\end{align}\\
Therefore the scheme to generate waiting times $W_j(t)$ is to first generate appropriate $W'$ from uniformly distributed random variates, and then to further tranform $W'$ by solving \ref{eq:inv_int}.\\
In practice, \ref{eq:inv_int} often cannot be solved analytically, and hence we resort to numeric methods.
\section{Inference Under Inhomogenous Coalescent}
\subsection{Exponential Growth}
\section{Coalescent with Local Population Structure}
\chapter{Results}
\section{Implementation Notes}
\section{Exponential Growth}
\subsection{Phylogeny Simulation}
\subsection{MCMC inference}
\section{Coalescent with Local Population Structure}
\subsection{Phylogeny Simulation}
\subsection{MCMC inference}
\chapter{Discussion}
\chapter{Bibliography}
\printbibliography
\end{document}
